bundle/src/main/java/com/adobe/acs/commons/mcp/impl/processes/asset/UrlAssetImport.java

```diff
@@ -0,0 +1,312 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2017 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.mcp.impl.processes.asset;
+
+import com.adobe.acs.commons.fam.ActionManager;
+import com.adobe.acs.commons.fam.actions.Actions;
+import com.adobe.acs.commons.functions.CheckedConsumer;
+import com.adobe.acs.commons.mcp.ProcessInstance;
+import com.adobe.acs.commons.mcp.form.FileUploadComponent;
+import com.adobe.acs.commons.mcp.form.FormField;
+import com.adobe.acs.commons.mcp.util.Spreadsheet;
+import com.day.cq.commons.jcr.JcrUtil;
+import com.day.cq.dam.api.Asset;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import javax.jcr.RepositoryException;
+import javax.jcr.Session;
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.client.HttpClient;
+import org.apache.http.osgi.services.HttpClientBuilderFactory;
+import org.apache.sling.api.request.RequestParameter;
+import org.apache.sling.api.resource.LoginException;
+import org.apache.sling.api.resource.ModifiableValueMap;
+import org.apache.sling.api.resource.PersistenceException;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.commons.mime.MimeTypeService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Import assets and metadata provided by a spreadsheet
+ */
+public class UrlAssetImport extends AssetIngestor {
+
+    public static String SOURCE = "source";
+    public static String TARGET_FOLDER = "target";
+    public static String ORIGINAL_FILE_NAME = "original";
+    public static String RENDITION_NAME = "rendition";
+    public static String CONTENT_BASE = "/content";
+    public static String UNKNOWN_TARGET_FOLDER = "/content/dam/unsorted";
+
+    private static final Logger LOG = LoggerFactory.getLogger(UrlAssetImport.class);
+    private HttpClient httpClient = null;
+
+    public UrlAssetImport(MimeTypeService mimeTypeService, HttpClientBuilderFactory httpFactory) {
+        super(mimeTypeService);
+        if (httpFactory != null) {
+            this.httpClient = httpFactory.newBuilder().build();
```

Comment:
I think this is where the timeouts need to be set.

Reply:
I can make it a configuration or something.  WDYT?

Reply:
configuration with defaults. See https://github.com/Adobe-Consulting-Services/acs-aem-commons/blob/master/bundle/src/main/java/com/adobe/acs/commons/http/impl/HttpClientFactoryImpl.java#L85

Reply:
```RequestConfig requestConfig = RequestConfig.custom()
                .setConnectTimeout(connectTimeout)
                .setSocketTimeout(soTimeout)
                .build();```

Got it... yeah that's what I did and moved this to init() after the timeout variable has populated from the deserialization logic.